### DISCUSSION TITLE
port in another branch interferes

### DISCUSSION THREAD

**Discussion by playgithub** [microsoft/vcpkg#24259] [General]
## Problem

In my private git branch, I've ported `basictoolbox`.
Now I want to port `wxcharts` publicly, steps as below:
1. new branch from `master` which keeps untouched
```sh
λ git switch master
λ git co -b port-wxcharts
```
2. port `wxcharts`
3. test
```sh
λ vcpkg install wxcharts
Computing installation plan...
Error: while loading basictoolbox:
The port directory (<vcpkg-root>\ports\basictoolbox) does not exist
Error: failed to load port from <vcpkg-root>\ports\basictoolbox
note: updating vcpkg by rerunning bootstrap-vcpkg may resolve this failure.
```

How can `basictoolbox` in another branch interfere with an irrelevant branch?
How to fix it? Tried `bootstrap-vcpkg`, but no help.

## More Info

Same error for branch `master`, e.g.
```sh
λ git switch master
λ vcpkg install somelib_in_master
Error: while loading basictoolbox:
...
```

**Comment by dg0yt** [comment]
This happens when an installed port doesn't exist in the vcpkg database. You can either remove it, or provide the port from another directory via overlay ports, or just work on a branch which includes the port.

**Reply by playgithub** [reply]
If installed ports and vcpkg database are always synchronized, it is better, right? Why not?

**Comment by playgithub** [comment]
To create a PR, a new branch without my private ports is needed.
Does it mean that my private ports has to be removed first? It's inconvenient.

**Reply by dg0yt** [reply]
A PR to vcpkg can't reference your private ports. So why not remove these packages as long as you work on another topic? Re-installation from artifact cache doesn't take much time. 
But you don't have to remove them. There are overlay ports, or leveraging git branches/tags/rebasing.

**Reply by playgithub** [reply]
OK, another choice is to clone a new repo for PR only.

**Reply by playgithub** [reply]
> A PR to vcpkg can't reference your private ports. So why not remove these packages as long as you work on another topic? Re-installation from artifact cache doesn't take much time.

If there are many private ports, it'll be not very easy.

> But you don't have to remove them. There are overlay ports, or leveraging git branches/tags/rebasing.

Overlay ports  introduces new folders, still not very easy.

In this case `basictoolbox` is irrelevant to `vcpkg install wxcharts`, why not check only the relevant things?
